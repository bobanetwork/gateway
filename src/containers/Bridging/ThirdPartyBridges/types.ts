export interface IThirdPartyBridge {
  name: string
  icon?: any
  iconLight?: any
  iconDark?: any
  link?: string
}
