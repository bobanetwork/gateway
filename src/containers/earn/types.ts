export enum toLayer {
  L1 = 'L2',
  L2 = 'L1',
}
