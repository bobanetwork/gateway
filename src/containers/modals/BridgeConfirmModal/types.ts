export type LayerType = 'L1' | 'L2'

export interface LayerIconProps {
  selectedLayer: LayerType
}

export interface ConfirmItemProps {
  label: string
  value: string | JSX.Element
}
