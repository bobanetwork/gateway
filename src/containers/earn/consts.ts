export const tableHeaderOptions = [
  { name: 'Token', width: 225 },
  {
    name: 'Available Balance',
    tooltip:
      'Available Balance refers to the amount of funds currently in each pool.',
    width: 145,
  },
  {
    name: 'Total Staked',
    tooltip: 'Total staked denotes the funds staked by liquidity providers.',
    width: 115,
  },
  {
    name: 'APR',
    tooltip:
      'The APR is the historical APR, which reflects the fees people paid to bridge and the previous usage patterns for each pool.',
    width: 85,
  },
  { name: 'Your Stake', width: 90 },
  { name: 'Earned', width: 110 },
  { name: 'Actions', width: 75 },
]
