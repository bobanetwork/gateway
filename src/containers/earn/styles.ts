import styled, { css } from 'styled-components'
import { Typography } from 'components/global/typography'

export const EarnPageContainer = styled.div`
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 10px;
  padding-top: 0;
  width: 1025px;
`

export const LayerAlert = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 30px;
  border-radius: 8px;
  margin: 8px 0;
  padding: 10px;
  background: ${(props) => props.theme.colors.box.background};
`

export const Help = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 30px;
  margin: 10px 0;
  padding: 10px;
  border-radius: 8px;
  background: ${(props) => props.theme.colors.box.background};
  ${({ theme: { name, colors } }) => css`
    border: 1px solid ${name === 'light' ? colors.blue[200] : colors.blue[100]};
    background: ${name === 'light' ? colors.blue[50] : colors.blue[500]};
    color: ${name === 'light' ? colors.blue[500] : colors.blue[100]};
  `}
`

export const AlertText = styled(Typography)`
  margin-left: 10px;
  flex: 4;
`

export const AlertInfo = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex: 1;
`

export const EarnAction = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
`

export const EarnActionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 10px 0;
`

export const EarnListContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px 0;
  padding: 10px 0;
`

export const TabSwitcherContainer = styled.div`
  display: flex;
  padding: 4px;
  gap: 0 15px;
  border-radius: 8px;
  ${({ theme: { colors, name } }) =>
    name === 'light'
      ? css`
          background: ${colors.gray[50]};
        `
      : css`
          background: ${colors.gray[500]};
        `}
`

export const Tab = styled.div<{ active: boolean }>`
  padding: 8px 24px;
  border-radius: 8px;
  cursor: pointer;
  ${(props) =>
    props.active &&
    `
        color:${props.theme.colors.gray[800]};
        background:${props.theme.colors.green[300]}
    `}
`
