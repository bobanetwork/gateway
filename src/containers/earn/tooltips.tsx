import React from 'react'
import { Typography } from 'components/global/typography'

export const BridgeTooltip = () => (
  <Typography variant="body3">
    <span style={{ fontWeight: '700' }}>Staking example</span>. When you stake
    10 BOBA into the L2 pool, then the pool's liquidity and balance both
    increase by 10 BOBA.
    <br />
    <br />
    <span style={{ fontWeight: '700' }}>Pool rebalancing</span>. In some
    circumstances, excess balances can accumulate on one chain. For example, if
    many people bridge from L1 to L2, then L1 pool balances will increase, while
    L2 balances will decrease. When needed, the pool operator can rebalance the
    pools, using 'classic' deposit and exit operations to move funds from one
    pool to another. Rebalancing takes 7 days, due to the 7 day fraud proof
    window, which also applies to the operator.
    <br />
    <br />
    <span style={{ fontWeight: '700' }}>Dynamic fees</span>. The pools use an
    automatic supply-and-demand approach to setting the fees. When a pool's
    liquidity is low, the fees are increased to attract more liquidity into that
    pool and vice-versa.
  </Typography>
)
