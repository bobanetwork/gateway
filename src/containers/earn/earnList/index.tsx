import { TableHeader } from 'components/global/table'
import React, { Fragment } from 'react'
import { useSelector } from 'react-redux'
import { selectPoolInfo, selectUserInfo } from 'selectors'
import { EarnListProps } from 'types/earn.types'
import { tableHeaderOptions } from '../table.header'
import EarnListItem from './earnListItem'
import { EarnListContainer, EearnListLoadingContainer } from './styles'
import { CircularProgress } from '@mui/material'

const EarnList = ({ lpChoice, showMyStakeOnly }: EarnListProps) => {
  const userInfo = useSelector(selectUserInfo())
  const poolInfo = useSelector(selectPoolInfo())
  const poolList: string[] = Object.keys(poolInfo[lpChoice])

  const isLoading = poolList.filter(Boolean).length === 0

  return (
    <>
      <TableHeader options={tableHeaderOptions} />
      {isLoading && (
        <EearnListLoadingContainer>
          <CircularProgress color="secondary" />
        </EearnListLoadingContainer>
      )}
      {!isLoading && (
        <>
          <EarnListContainer>
            {poolList.map((tokenAddress, i) => {
              return (
                <Fragment key={i}>
                  <EarnListItem
                    showMyStakeOnly={showMyStakeOnly}
                    poolInfo={poolInfo[lpChoice][tokenAddress]}
                    userInfo={userInfo[lpChoice][tokenAddress]}
                    lpChoice={lpChoice}
                    tokenAddress={tokenAddress}
                  />
                </Fragment>
              )
            })}
          </EarnListContainer>
        </>
      )}
    </>
  )
}

export default React.memo(EarnList)
