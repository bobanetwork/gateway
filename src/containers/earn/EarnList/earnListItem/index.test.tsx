import {
  fireEvent,
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import networkService from 'services/networkService'
import CustomThemeProvider from 'themes'
import { LiquidityPoolLayer } from 'types/earn.types'
import { mockedInitialState } from 'util/tests'
import EarnListItem, { EarnListItemProps } from '.'

const mockStore = configureStore([thunk])

interface TestEarnListItemProps extends EarnListItemProps {
  accountEnabled?: boolean
  store?: any
}

jest.mock('services/networkService', () => {
  return {
    getL1LPInfo: jest.fn(),
    getL2LPInfo: jest.fn(),
    getReward: jest.fn(),
  }
})

const renderEarnListItemComponent = ({
  poolInfo,
  userInfo,
  chainId,
  tokenAddress,
  lpChoice,
  showMyStakeOnly,
  store,
}: TestEarnListItemProps) => {
  return render(
    <Provider store={store}>
      <BrowserRouter>
        <CustomThemeProvider>
          <EarnListItem
            poolInfo={poolInfo}
            userInfo={userInfo}
            chainId={chainId}
            tokenAddress={tokenAddress}
            lpChoice={lpChoice}
            showMyStakeOnly={showMyStakeOnly}
          />
        </CustomThemeProvider>
      </BrowserRouter>
    </Provider>
  )
}

const TokenAddress = '0x429582bde1b0e011c48d883354050938f194743f'

describe('EarnList ', () => {
  let lpChoice: any
  let showMyStakeOnly: boolean
  let tokenAddress: string
  let chainId: any
  let store

  let userInfo: any = {
    l2TokenAddress: '0x429582bde1b0e011c48d883354050938f194743f',
    amount: '0',
    pendingReward: '0',
    rewardDebt: '0',
  }

  let poolInfo: any = {
    symbol: 'USDC',
    name: 'USD Coin',
    decimals: 6,
    l1TokenAddress: '0x07865c6e87b9f70255377e024ace6630c1eaa37f',
    l2TokenAddress: '0x429582bde1b0e011c48d883354050938f194743f',
    accUserReward: '0',
    accUserRewardPerShare: '0',
    userDepositAmount: '0',
    startTime: '1668555514',
    APR: 0,
    tokenBalance: '0',
  }

  beforeEach(() => {
    lpChoice = LiquidityPoolLayer.L1LP
    tokenAddress = TokenAddress
    showMyStakeOnly = false
    store = mockStore({
      ...mockedInitialState,
    })
    // @ts-ignore
    networkService.getL1LPInfo.mockImplementation(() =>
      Promise.resolve({ poolInfo: {}, userInfo: {} })
    )
    // @ts-ignore
    networkService.getL2LPInfo.mockImplementation(() =>
      Promise.resolve({ poolInfo: {}, userInfo: {} })
    )
  })

  test('should match snapshot when showMystake is disabled', () => {
    showMyStakeOnly = false
    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshot when showMystake is enable', () => {
    showMyStakeOnly = true
    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshot when lpChoice empty', () => {
    lpChoice = ''
    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshot when lpChoice L1LP', () => {
    lpChoice = LiquidityPoolLayer.L1LP
    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshot when lpChoice L2LP', () => {
    lpChoice = LiquidityPoolLayer.L2LP
    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshot when userreward is available', () => {
    showMyStakeOnly = true
    userInfo = {
      l2TokenAddress: '0x4200000000000000000000000000000000000023',
      amount: '43006000000000000000',
      pendingReward: '1100191312876000',
      rewardDebt: '27734072508676000',
    }

    poolInfo = {
      symbol: 'BOBA',
      name: 'Boba Token',
      decimals: 18,
      l1TokenAddress: '0xeccd355862591cbb4bb7e7dd55072070ee3d0fc1',
      l2TokenAddress: '0x4200000000000000000000000000000000000023',
      accUserReward: '24767238000000000',
      accUserRewardPerShare: '648705579',
      userDepositAmount: '565303100000000000000',
      startTime: '1668555549',
      APR: 0.004145006519156489,
      tokenBalance: '614833937808000000000',
    }

    store = mockStore({
      ...mockedInitialState,
      setup: {
        ...mockedInitialState.setup,
        accountEnabled: true,
      },
    })

    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot when userInfo or poolInfo is empty', () => {
    showMyStakeOnly = true
    userInfo = {}

    poolInfo = {}

    const { asFragment } = renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should show and hide styledMenu on click of actionBtn', async () => {
    renderEarnListItemComponent({
      lpChoice,
      showMyStakeOnly,
      userInfo,
      poolInfo,
      tokenAddress,
      chainId,
      store,
    })

    fireEvent.click(screen.getByTestId('action-btn'))
    expect(screen.getByText('Unstake')).toBeInTheDocument()
    fireEvent.click(screen.getByRole('presentation') as any)
    await waitForElementToBeRemoved(() => screen.queryByText('Unstake'))
  })

  describe('Handle unstake token', () => {
    test('should dispatch state with correct payload incase of L1LP ', () => {
      renderEarnListItemComponent({
        lpChoice,
        showMyStakeOnly,
        userInfo,
        poolInfo,
        tokenAddress,
        chainId,
        store,
      })

      fireEvent.click(screen.getByTestId('action-btn'))
      expect(screen.getByText('Unstake')).toBeInTheDocument()

      fireEvent.click(screen.getByTestId('Unstake'))

      const actions = store.getActions()
      expect(actions).toContainEqual({
        payload: {
          L1orL2Pool: 'L1LP',
          LPAddress: undefined,
          balance: 0,
          currency: undefined,
          decimals: undefined,
          symbol: undefined,
        },
        type: 'UPDATE_WITHDRAW_TOKEN',
      })
      expect(actions).toContainEqual({
        destNetworkSelection: undefined,
        fast: undefined,
        lock: undefined,
        payload: 'EarnWithdrawModal',
        proposalId: undefined,
        selectionLayer: undefined,
        token: undefined,
        tokenIndex: undefined,
        type: 'UI/MODAL/OPEN',
      })
    })
    test('should dispatch state with correct payload incase of L2LP ', () => {
      lpChoice = LiquidityPoolLayer.L2LP
      renderEarnListItemComponent({
        lpChoice,
        showMyStakeOnly,
        userInfo,
        poolInfo,
        tokenAddress,
        chainId,
        store,
      })

      fireEvent.click(screen.getByTestId('action-btn'))
      expect(screen.getByText('Unstake')).toBeInTheDocument()

      fireEvent.click(screen.getByTestId('Unstake'))

      const actions = store.getActions()
      expect(actions).toContainEqual({
        payload: {
          L1orL2Pool: 'L2LP',
          LPAddress: undefined,
          balance: '00',
          currency: undefined,
          decimals: undefined,
          symbol: undefined,
        },
        type: 'UPDATE_WITHDRAW_TOKEN',
      })
      expect(actions).toContainEqual({
        destNetworkSelection: undefined,
        fast: undefined,
        lock: undefined,
        payload: 'EarnWithdrawModal',
        proposalId: undefined,
        selectionLayer: undefined,
        token: undefined,
        tokenIndex: undefined,
        type: 'UI/MODAL/OPEN',
      })
    })
    test('should dispatch state with correct payload incase of empty ', () => {
      lpChoice = ''
      renderEarnListItemComponent({
        lpChoice,
        showMyStakeOnly,
        userInfo,
        poolInfo,
        tokenAddress,
        chainId,
        store,
      })

      fireEvent.click(screen.getByTestId('action-btn'))
      expect(screen.getByText('Unstake')).toBeInTheDocument()

      fireEvent.click(screen.getByTestId('Unstake'))

      const actions = store.getActions()
      expect(actions).toContainEqual({
        payload: {
          L1orL2Pool: '',
          LPAddress: undefined,
          balance: 0,
          currency: undefined,
          decimals: undefined,
          symbol: undefined,
        },
        type: 'UPDATE_WITHDRAW_TOKEN',
      })
      expect(actions).toContainEqual({
        destNetworkSelection: undefined,
        fast: undefined,
        lock: undefined,
        payload: 'EarnWithdrawModal',
        proposalId: undefined,
        selectionLayer: undefined,
        token: undefined,
        tokenIndex: undefined,
        type: 'UI/MODAL/OPEN',
      })
    })
  })
  describe('Handle harvest', () => {
    test('should dispatch state with correct payload incase of L1LP ', () => {
      renderEarnListItemComponent({
        lpChoice,
        showMyStakeOnly,
        userInfo,
        poolInfo,
        tokenAddress,
        chainId,
        store,
      })

      fireEvent.click(screen.getByTestId('action-btn'))
      expect(screen.getByText('Harvest')).toBeInTheDocument()

      fireEvent.click(screen.getByTestId('Harvest'))

      const actions = store.getActions()
      expect(actions).toContainEqual({ type: 'EARN/HARVEST/REQUEST' })
    })
    test('should dispatch state with correct payload incase of L2LP ', () => {
      lpChoice = LiquidityPoolLayer.L2LP
      renderEarnListItemComponent({
        lpChoice,
        showMyStakeOnly,
        userInfo,
        poolInfo,
        tokenAddress,
        chainId,
        store,
      })

      fireEvent.click(screen.getByTestId('action-btn'))
      expect(screen.getByText('Harvest')).toBeInTheDocument()

      fireEvent.click(screen.getByTestId('Harvest'))

      const actions = store.getActions()
      expect(actions).toContainEqual({ type: 'EARN/HARVEST/REQUEST' })
    })
  })
})
