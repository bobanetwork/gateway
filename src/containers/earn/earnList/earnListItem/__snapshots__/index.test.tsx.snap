// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarnList  should match snapshot when lpChoice L1LP 1`] = `
<DocumentFragment>
  <div
    class="sc-dAbbOL kfujas"
    data-testid="earn-list-item"
  >
    <div
      class="sc-aXZVg krHmEs sc-kpDqfm hvflIl"
    >
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 225px;"
      >
        <div
          class="sc-aXZVg krHmEs sc-iGgWBj xSUkw"
        >
          <div
            class="sc-gEvEer Potal"
          >
            <div
              class="sc-aXZVg krHmEs"
            >
              <div>
                <img
                  alt="logo"
                  class="sc-dcJsrY khyFAh"
                  src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/usdc.svg"
                />
              </div>
              <div>
                <p
                  class="sc-eqUAAy ibYEEy"
                >
                  USDC
                </p>
                <p
                  class="sc-fqkvVR jpBRQY"
                >
                  USD Coin
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 145px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 115px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 85px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 90px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 110px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 75px;"
      >
        <div
          class="sc-feUZmu ceIvoQ"
          data-testid="action-btn"
        >
          <div
            class="sc-fUnMCh euYGlO"
          >
            <div>
              <div
                data-src="actions.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EarnList  should match snapshot when lpChoice L2LP 1`] = `
<DocumentFragment>
  <div
    class="sc-dAbbOL kfujas"
    data-testid="earn-list-item"
  >
    <div
      class="sc-aXZVg krHmEs sc-kpDqfm hvflIl"
    >
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 225px;"
      >
        <div
          class="sc-aXZVg krHmEs sc-iGgWBj xSUkw"
        >
          <div
            class="sc-gEvEer Potal"
          >
            <div
              class="sc-aXZVg krHmEs"
            >
              <div>
                <img
                  alt="logo"
                  class="sc-dcJsrY khyFAh"
                  src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/usdc.svg"
                />
              </div>
              <div>
                <p
                  class="sc-eqUAAy ibYEEy"
                >
                  USDC
                </p>
                <p
                  class="sc-fqkvVR jpBRQY"
                >
                  USD Coin
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 145px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 115px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 85px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 90px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 110px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 75px;"
      >
        <div
          class="sc-feUZmu ceIvoQ"
          data-testid="action-btn"
        >
          <div
            class="sc-fUnMCh euYGlO"
          >
            <div>
              <div
                data-src="actions.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EarnList  should match snapshot when lpChoice empty 1`] = `
<DocumentFragment>
  <div
    class="sc-dAbbOL kfujas"
    data-testid="earn-list-item"
  >
    <div
      class="sc-aXZVg krHmEs sc-kpDqfm hvflIl"
    >
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 225px;"
      >
        <div
          class="sc-aXZVg krHmEs sc-iGgWBj xSUkw"
        >
          <div
            class="sc-gEvEer Potal"
          >
            <div
              class="sc-aXZVg krHmEs"
            >
              <div>
                <img
                  alt="logo"
                  class="sc-dcJsrY khyFAh"
                  src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/usdc.svg"
                />
              </div>
              <div>
                <p
                  class="sc-eqUAAy ibYEEy"
                >
                  USDC
                </p>
                <p
                  class="sc-fqkvVR jpBRQY"
                >
                  USD Coin
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 145px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 115px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 85px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 90px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 110px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 75px;"
      >
        <div
          class="sc-feUZmu ceIvoQ"
          data-testid="action-btn"
        >
          <div
            class="sc-fUnMCh euYGlO"
          >
            <div>
              <div
                data-src="actions.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EarnList  should match snapshot when showMystake is disabled 1`] = `
<DocumentFragment>
  <div
    class="sc-dAbbOL kfujas"
    data-testid="earn-list-item"
  >
    <div
      class="sc-aXZVg krHmEs sc-kpDqfm hvflIl"
    >
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 225px;"
      >
        <div
          class="sc-aXZVg krHmEs sc-iGgWBj xSUkw"
        >
          <div
            class="sc-gEvEer Potal"
          >
            <div
              class="sc-aXZVg krHmEs"
            >
              <div>
                <img
                  alt="logo"
                  class="sc-dcJsrY khyFAh"
                  src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/usdc.svg"
                />
              </div>
              <div>
                <p
                  class="sc-eqUAAy ibYEEy"
                >
                  USDC
                </p>
                <p
                  class="sc-fqkvVR jpBRQY"
                >
                  USD Coin
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 145px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 115px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 85px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 90px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 110px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 75px;"
      >
        <div
          class="sc-feUZmu ceIvoQ"
          data-testid="action-btn"
        >
          <div
            class="sc-fUnMCh euYGlO"
          >
            <div>
              <div
                data-src="actions.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EarnList  should match snapshot when showMystake is enable 1`] = `<DocumentFragment />`;

exports[`EarnList  should match snapshot when userInfo or poolInfo is empty 1`] = `
<DocumentFragment>
  <div
    class="sc-dAbbOL kfujas"
    data-testid="earn-list-item"
  >
    <div
      class="sc-aXZVg krHmEs sc-kpDqfm hvflIl"
    >
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 225px;"
      >
        <div
          class="sc-aXZVg krHmEs sc-iGgWBj xSUkw"
        >
          <div
            class="sc-gEvEer Potal"
          >
            <div
              class="sc-aXZVg krHmEs"
            >
              <div>
                <img
                  alt="logo"
                  class="sc-dcJsrY khyFAh"
                  src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/undefined.svg"
                />
              </div>
              <div>
                <p
                  class="sc-eqUAAy ibYEEy"
                />
                <p
                  class="sc-fqkvVR jpBRQY"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 145px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 115px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 85px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          NaN
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 90px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 110px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 75px;"
      >
        <div
          class="sc-feUZmu ceIvoQ"
          data-testid="action-btn"
        >
          <div
            class="sc-fUnMCh euYGlO"
          >
            <div>
              <div
                data-src="actions.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EarnList  should match snapshot when userreward is available 1`] = `
<DocumentFragment>
  <div
    class="sc-dAbbOL kfujas"
    data-testid="earn-list-item"
  >
    <div
      class="sc-aXZVg krHmEs sc-kpDqfm hvflIl"
    >
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 225px;"
      >
        <div
          class="sc-aXZVg krHmEs sc-iGgWBj xSUkw"
        >
          <div
            class="sc-gEvEer Potal"
          >
            <div
              class="sc-aXZVg krHmEs"
            >
              <div>
                <img
                  alt="logo"
                  class="sc-dcJsrY khyFAh"
                  src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/boba.svg"
                />
              </div>
              <div>
                <p
                  class="sc-eqUAAy ibYEEy"
                >
                  BOBA
                </p>
                <p
                  class="sc-fqkvVR jpBRQY"
                >
                  Boba Token
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 145px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          614.83
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 115px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          565.3
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 85px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 90px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          43.01
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 110px;"
      >
        <p
          class="sc-imWYAI dZiXLh"
        >
          0.00126
        </p>
      </div>
      <div
        class="sc-aXZVg krHmEs sc-dAlyuH iRhOjN"
        style="max-width: 75px;"
      >
        <div
          class="sc-feUZmu ceIvoQ"
          data-testid="action-btn"
        >
          <div
            class="sc-fUnMCh euYGlO"
          >
            <div>
              <div
                data-src="actions.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
